packer/resource-ami/init.sh

```diff
@@ -0,0 +1,125 @@
+#!/bin/bash
+
+# Copyright (c) 2018, WSO2 Inc. (http://wso2.com) All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+ubuntu_ami="ami-0584bcdcc2b2a82e9"
+ubuntu_os_version="16.04"
+ubuntu_ssh_username="ubuntu"
+
+centos_ami="ami-9887c6e7"
+centos_os_version="7.4"
+centos_ssh_username="centos"
+
+packer_file="packer-conf.json"
+
+#=== FUNCTION ==================================================================
+# NAME: startPacker
+# DESCRIPTION: Validate config json and start packer.
+#===============================================================================
+function startPacker() {
+	packer validate packer-conf.json
+	packer build packer-conf.json
+}
+
+#=== FUNCTION ==================================================================
+# NAME: checkIfExists
+# DESCRIPTION: Check if the provided file exists.
+# PARAMETER 1: File to check 
+#===============================================================================
+function checkIfExists() {
+	file=$1
+	if [ -f "$file" ]
+	then
+		echo "$file found."
+		return 0
+	else
+		echo "$file not found."
+		return 1
+	fi
+}
+
+#=== FUNCTION ==================================================================
+# NAME: checkResources
+# DESCRIPTION: Check if the resources given in resource.txt exists.
```

Comment:
What is a resource, and what it needs to contain? Is there a sample?

Reply:
All the items that we keep in S3 to be used with packer are consider as resources.
For an example for CentOS we have:
1. agent.zip (TestGrid tinkerer agent)
2. perf_monitoring_artifacts.zip (Jolokia config stuff)
3. oracle db-client (After agreeing on EULA)
If you can remember that we sync 3 places in S3 to collect resources we need to add to CentOS as;
testgrid-resources/packer/Common
testgrid-resources/packer/Unix
testgrid-resources/packer/CentOS

So there is tendency to lost some resources. (Developer may not add the resource hoping that it is existing in one of the syncing locations.). Therefore we use a resources.txt file in each OS-specific location (ex: testgrid-resources/packer/**CentOS**) which has all the necessary resources needed end of the day.
So we check if all they are available once after syncing the 3 locations using the resources.txt file.

Reply:
Example of a resource.txt file (A list of resources):

```
ojdbc8.jar
agent.zip
oracle-instantclient12.2-basic-12.2.0.1.0-1.x86_64.rpm
oracle-instantclient12.2-sqlplus-12.2.0.1.0-1.x86_64.rpm
```